### ISSUE TITLE
Invalid icon size causing icons not to display

### ISSUE THREAD

**Issue by lazytanuki** [phuhl/linux_notification_center#33]
Hello there !

I'd like to begin by showing my appreciation for your work, I really like what you did !

Now for the issue : when I get a Telegram notification (native), the notifications buttons show broken icons (both in popups and in center) while the program gives this :  

`(deadd-notification-center:255448): Gtk-WARNING **: 03:07:41.477: Invalid icon size 10`

I have got this from dbus-monitor if it can help (it is the Telegram notification) : 

```
   string "Telegram Desktop"
   uint32 0
   string ""
   string "Test"
   string "Test Notification
This is a test message"
   array [
      string "default"
      string ""
      string "mail-mark-read"
      string "Mark as read"
      string "mail-reply-sender"
      string "Reply"
   ]
   array [
      dict entry(
         string "action-icons"
         variant             boolean true
      )
      dict entry(
         string "category"
         variant             string "im.received"
      )
      dict entry(
         string "desktop-entry"
         variant             string "telegramdesktop"
      )
```

The two icons here are "mail-mark-read" and "mail-reply-sender". I have checked that I have them (I have a common Papirus icon theme). It though seems that all icons start from size 16, which could explain why GTK can't print it in size 10.

I have looked for a way to set this value in the configuration file but did not succeed. Or maybe there could be a way to completely ignore button icons ?

Thank you for your time !

**Comment by phuhl** [comment]
Hey there, thank you for the praise and the bug report.

So sorry, for taking so long to reply. I think, I found your error:

https://github.com/phuhl/linux_notification_center/blob/1f20696d152f7359cbdbcfb4e997f00135c673e1/src/NotificationCenter/Notifications/Action.hs#L66

For the button icons the size is hardcoded to 10. This should be a configurable option, I guess.

Also, this issue has been rereported here: #62. You are not the only one, experiencing this issue. A second solution could be, to rewrite the icons as suggested here: #68.

**Comment by MyriaCore** [comment]
I've run into this bug when using [`imageNewFromIconName`](https://hackage.haskell.org/package/gi-gtk-3.0.26/docs/GI-Gtk-Objects-Image.html#v:imageNewFromIconName) before, although I've not had it refuse to display my icons because of it. I ran into this when resolving [an issue with my taffybar](https://gitlab.com/myriacore-dotfiles/taffybar/-/issues/1). The GTK equivalent is [`gtk_image_new_from_icon_name`](https://developer.gnome.org/gtk3/unstable/GtkImage.html#gtk-image-new-from-icon-name). In place of  where we have`10`, this function takes a [`GtkIconSize`](https://developer.gnome.org/gtk3/unstable/gtk3-Themeable-Stock-Images.html#GtkIconSize), which is an enum. 

While you *could* just pass in an int value, you might just want to use [`GI.Gtk.Enums.IconSize(..)`](https://hackage.haskell.org/package/gi-gtk-3.0.26/docs/GI-Gtk-Enums.html#t:IconSize) instead.

**Comment by MyriaCore** [comment]
@lazytanuki this is closed, but feel free to re-open or chime in over at #62 if this didn't actually resolve this bug.

**Comment by tpurde** [comment]
@MyriaCore has this been merged?

**Comment by MyriaCore** [comment]
@tpurde It has, but the fix hasn't been deployed to AUR yet. If you *are* an arch user, here are your workarounds:
- Track #90 and wait
- build from the source. You may run into #85/#45, though
- I've built the binary and included it myself in #92, so you can download that, unzip it, and run `sudo make install` in the project directory. You shouldn't need stack or haskell deps to do this.

**Comment by tpurde** [comment]
> yes

Thank you.

@MyriaCore I think I'll just wait for #92 to be merged.
